### PR TITLE
Implement posts feature

### PR DESCRIPTION
**Pull request by pbkabali**
### What this PR does:
- Adds post model, controllers, and views
- Adds user authentication using Devise
- Adds view logic to distinguish between users who are logged in or not

### CODE REVIEW

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @pbkabali and @alvp01 
### You have done well so far :clap: However, you have a few issues to fix:-
- [x] .stylelintrc.json file is missing as it is a requirement, kindly add it. [Follow this guide](https://github.com/microverseinc/linters-config/tree/master/ror)
### Make the changes and submit for another code review. Good luck.
- Happy coding!

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  validates :username, :email, :password, presence: true
+  validates :username, :email, uniqueness: true
+  validates :username, :password, length: { in: 4..10 }
```
Comment (rOluochKe) [inline app/models/user.rb]: ### This double validation, devise already has this. Therefore, delete it

**Comment by pbkabali** [comment]
Thanks for the review @rOluochKe. We have made the changes and submitted another review request.

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️
### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Comment by pbkabali** [comment]
Thank you for your review @voscarmv

**Pull request merged**